### ISSUE TITLE
Splitting core functionality to separate plugins

### ISSUE THREAD

**Issue by KELiON** [KELiON/cerebro#277]
Hello, dear contributors and Cerebro users!

In this issue I want to raise awareness and hear your feedback about my thoughts. I see that complexity and amount of core features are growing and now it is a time to split everything and leave only *really* core features in the main repository and move other stuff to separate packages. For that purpose I've already created repositories for everything, that I think should be moved out, such as:

* [x] https://github.com/KELiON/cerebro-ui – here I want to create another package, like `cerebro-tools`, but with common react-components, that could be reused in plugins. It will help to have consistent UI across all plugins

Repositories for plugins:

* [x] https://github.com/KELiON/cerebro-basic-apps
* [x] https://github.com/KELiON/cerebro-converter
* [x] https://github.com/KELiON/cerebro-files-nav
* [x] https://github.com/KELiON/cerebro-google
* [x] https://github.com/KELiON/cerebro-google-maps
* [x] https://github.com/KELiON/cerebro-math
* [x] https://github.com/KELiON/cerebro-open-web
* [x] https://github.com/KELiON/cerebro-mac-apps
* [x] https://github.com/KELiON/cerebro-yandex-translate

Moving all plugins to separate repositories/packages will help to keep them small, clear, and testable. It will reduce size of application installer. Also, it will help to involve more contributors, because some developers could be more interested in helping with a specific plugin instead of dealing with huge amount of code and building process of main application.

At the same time it helps us with UX, because after that change every plugin in the system could be installed or uninstalled by the user (so, i.e. currently some users already uses duck-duck-go search, but disabling google is not possible). I thought about adding "disable" option, but why should we keep plugins, that are not used?

So, I want to track here updates of moving code out. Contributors are welcome to help – it will increase our speed of doing it! Just leave a comment if you start  to extracting any plugin.

As a last step here we will have to add initial installing of some plugins and we have to decide which plugins should be included as core. As I see it, it will be:

* basic-apps (apps search and launch on windows and linux)
* osx-app 
* math
* converter
* files navigation
* google
* open web

So, basically translate and google maps plugins could be removed from core. Share you thoughts in comments!

**Comment by KELiON** [comment]
Starting to extract converter plugin

**Comment by maximbaz** [comment]
This will be a great improvement :+1: 

Let's also keep in mind what we discussed in #118, namely not having a second BrowserWindow for background work. This refactoring doesn't have to happen immediately, but let's design this whole process in a way that every single plugin is expected to create its own lighweight `child_process` to handle its work, Cerebro itself should not keep around a (potentially unused) second BrowserWindow for such plugins.

**Comment by KELiON** [comment]
> that every single plugin is expected to create its own lighweight child_process to handle its work
I'd expect that no one will do it:) We should create simple API for background processing and describe cases when it should be used. Otherwise just everyone will use main process, that will slow down overall performance.

**Comment by glja021** [comment]
This looks like a great change :+1: 
It'd make developing new features a lot easier and quicker.

Tonight I'm going to extract the Google plugin :smile: 
Starting the math after that one is done

**Comment by glja021** [comment]
So I've got them both working by themselves and started looking at the files-nav one, looks a bit more complex with the async calls. Maybe it'd make sense to extract the rpc's functions into another package that handles the background tasks?

By the way I couldn't send a pull request for the plugins as your repos are currently empty, they are [here](https://github.com/glja021/cerebro-google) and [here](https://github.com/glja021/cerebro-math).

**Comment by KELiON** [comment]
@glja021 awesome, thanks! I've given you push access to my repos.

**Comment by KELiON** [comment]
Going to extract google maps now

**Comment by maximbaz** [comment]
This deserves some extra praise, very nice improvement! :+1: 🎉 
Hoping to see 0.2.9 released soon 😜
